### PR TITLE
More flexible dependency versions

### PR DESCRIPTION
**Pull request by benlangfeld**
Allows consuming applications to upgrade all dependencies within the specified major version, so as not to block feature and bug fix upgrades on playbook releases.

Since this does not permit major version upgrades, and thus backward incompatible upgrades, it should not be problematic.

### CODE REVIEW

File: playbook/package.json
Comment (thestephenmarshall) [inline playbook/package.json:93]: Realistically, most of these should be `devDependencies` which means you shouldn't care at all about the version. The only thing we ship non-transpiled is our SCSS tokens which are pretty standard and shouldn't really care about the version of `node-sass` either.

File: playbook/package.json
Comment (thestephenmarshall) [inline playbook/package.json:118]: We should be careful here as what happens here is we tell the transpiler *not* to include this version, but instead require that the consumer of the library provide the version specified in their application. We are fairly confident that much later versions of React would still be backwards compatible with the transpiled JavaScript, but this is definitely a place that needs much more testing. `react` and `react-dom` as you can imagine are the main libs of concern here.

**Review by thestephenmarshall — CHANGES_REQUESTED** [review]
This needs alpha testing before we can move any further with this PR. We cannot ship this before that has been done. Also, I'll defer to the opinions of the rest of @powerhome/ux-team  with regards to the severity of the changes here and whether or not this indicates the need for a major release or not. It is my opinion that this could possibly be shipped as a regular release. Again, this depends on alpha testing against this PR alone first.

File: playbook/package.json
Comment (benlangfeld) [reply playbook/package.json:93]: I'm all for moving things to devDependencies. If you can list which ones can move and which have to stay, I'm totally up for that.

**Comment by benlangfeld** [comment]
If this whole change would be of concern, but there similar relaxation of dependencies (including the above suggestion to make many of these `devDependencies`) that the UX team can provide, @powerhome/eternals would be very grateful; they have a list of open PRs which are marked as blocked on playbook for this reason.

File: playbook/package.json
Comment (thestephenmarshall) [reply playbook/package.json:93]: I would say everything here in `package.json` can be moved to devDependencies as the only ones we are truly concerned about are in peerDependencies currently. These being in the wrong place has been a matter of contributors not using the `-D` flag when installing a library and/or not understanding what goes where and why.

**Comment by thestephenmarshall** [comment]
I can add a commit here to move things to where we see fit. Is that okay @benlangfeld ?

**Comment by benlangfeld** [comment]
@thestephenmarshall All yours :)

**Comment by benlangfeld** [comment]
Hey @thestephenmarshall , it seems we have a Nitro PR open to test this. Are we going this way, or moving those deps to `devDependencies`?

**Comment by thestephenmarshall** [comment]
@benlangfeld that is correct. I have moved most everything to devDependencies in my last commit and fixed other items that make our dev docs in Nitro work for the React side as well.

**Review by thestephenmarshall — APPROVED** [review]

**Comment by benlangfeld** [comment]
Thanks @thestephenmarshall , I'm looking forward to this.

**Comment by thestephenmarshall** [comment]
@benlangfeld the only dependency that I had to end up specifying was `webpack` because [it is a webpack-dev-server peerDependency](https://github.com/webpack/webpack-dev-server/blob/master/package.json#L116)

**Comment by benlangfeld** [comment]
Thanks for getting this tested and merged. Please let me know when it makes it into a release so we can upgrade Tempo.

**Comment by thestephenmarshall** [comment]
@benlangfeld it has been yanked

**Comment by thestephenmarshall** [comment]
we need to see why CI is being a PITA

**Comment by thestephenmarshall** [comment]
[New PR](https://github.com/powerhome/playbook/pull/1440)

**Pull request merged**
